syntax = "proto3";

package protos;



import "google/protobuf/empty.proto";
import "custom_types.proto";

service CustomerAddressService {
  rpc GetAll (GetAllRequest) returns (GetAllCustomerAddressesResponse);
  rpc GetById (GetCustomerAddressByIdRequest) returns (CustomerAddressModel);
}

message CustomerAddressModel {
  int32 address_id = 1;
  optional int32 customer_id = 2;
  optional string country = 3;
  optional string city = 4;
  optional string street = 5;
  optional string zip_code = 6;
  optional bool is_primary = 7;
}

message GetAllCustomerAddressesResponse {
  repeated CustomerAddressModel customer_addresses = 1;
}

message GetCustomerAddressByIdRequest {
  int32 address_id = 1;
}
